### DISCUSSION TITLE
Reordable rows on dragn drop

### DISCUSSION THREAD

**Discussion by rudolfbruder** [rappasoft/laravel-livewire-tables#308] [Feature Requests]
Hello,

It would be awesome for this package to support[ livewire sortable](https://github.com/livewire/sortable) package. I managed to get it working but i need to add public $orderable property in the main component in order to access it in table components in blade views. Right now i am faking it with true false set in the table blade component.

I would switch sortable naming to orderable so it does not messup this compoenent sortable functions.

I modified tbody and row components to apply the orderable actions if the orderable property is set to true. 

![image](https://user-images.githubusercontent.com/56951790/120346697-e76a1100-c2fb-11eb-9251-6fb0ae8c667a.png)

![image](https://user-images.githubusercontent.com/56951790/120346776-f8b31d80-c2fb-11eb-8804-f7e5dac43135.png)

![image](https://user-images.githubusercontent.com/56951790/120346839-08cafd00-c2fc-11eb-92ef-209104d62496.png)

Do you think that you could approve it?

**Comment by rappasoft** [comment]
I actually have a fully working branch of livewire sortable sorting rows and calling a user-defined update method. However, it only works on a per-page basis, so if you sort rows on one page and switch pages and sort rows on another page, it doesn't work because the indexes are reset to 0-N (per page). The only way it really makes sense would be to only use it when you are looking at all rows at a time. That and the amount of code that I added was a lot to maintain vs. the number of people that would probably use the feature. So I ended up dumping it.

**Reply by rudolfbruder** [reply]
I understand, i would definitely use some single property which could be adjustable by end users and accessible in our components which extend the DataTableComponent. By doing this we could customize the blade templates so that it would work for us. Do you think you could adjust it as i suggested in pull request? I mean it doesnt have to be called sortable it can just be called somewhat generally. But it would help if we could access properties setup on our components in the default blade components of this package.

**Reply by rappasoft** [reply]
I have to think about the 50k people that use this plugin as 99% don't need this feature. There should be nothing stopping you from publishing the views and adding what you have above and adding that method to your component, it should all work.

**Reply by rudolfbruder** [reply]
We have tried that with the team and the problem was that when we declare some public property such as $orderable in our component we were not able to access it in the table.blade files. It was accessible only if we faked the property in the vendor file DataTableComponent.php. Maybe we are doing something wrong...

**Reply by rappasoft** [reply]
That might be a valid point as I have never tried it and I guess no one else has either. I'll have to look into it.

**Comment by lrljoe** [comment]
This is present in v2+
